api-samples/tabs/inspector/tabs_api.js

```diff
@@ -0,0 +1,403 @@
+  windowItem.querySelector('.window_width').id = `width_${window.id}`;
+  windowItem.querySelector('.window_height').id = `height_${window.id}`;
+  windowItem.querySelector('.window_focused').id = `focused_${window.id}`;
+  windowItem.querySelector('.window_current').id = `current_${window.id}`;
+  windowItem.querySelector('.window_id').innerText = window.id;
+  windowItem.querySelector('.window_left').value = window.left;
+  windowItem.querySelector('.window_top').value = window.top;
+  windowItem.querySelector('.window_width').value = window.width;
+  windowItem.querySelector('.window_height').value = window.height;
+  windowItem.querySelector('.window_focused').checked = window.focused;
+  windowItem.querySelector('.window_current').checked = window.current;
+
+  windowItem.querySelector('.window_refresh').addEventListener('click', () => {
+    refreshWindow(window.id);
+  });
+
+  windowItem
+    .querySelector('.update_window_button')
+    .addEventListener('click', () => {
+      updateWindow(window.id);
+    });
+
+  windowItem
+    .querySelector('.remove_window_button')
+    .addEventListener('click', () => {
+      removeWindow(window.id);
+    });
+
+  windowItem
+    .querySelector('.refresh_selected_tab_button')
+    .addEventListener('click', () => {
+      refreshSelectedTab(window.id);
+    });
+
+  windowItem.querySelector('#tabList').innerHTML = '';
+  for (let tab of window.tabs) {
+    const tabItem = document.importNode(tabTemplate, true);
+    renderTab(tab, tabItem.children[0]);
+    windowItem.querySelector('#tabList').appendChild(tabItem);
+  }
+}
+
+function renderTab(tab, tabItem) {
+  tabItem.id = `tab_${tab.id}`;
+  tabItem.querySelector('.tab_index').id = `index_${tab.id}`;
+  tabItem.querySelector('.tab_window_id').id = `windowId_${tab.id}`;
+  tabItem.querySelector('.tab_title').id = `title_${tab.id}`;
+  tabItem.querySelector('.tab_url').id = `url_${tab.id}`;
+  tabItem.querySelector('.tab_selected').id = `selected_${tab.id}`;
+
+  tabItem.querySelector('.tab_id').innerText = `TabId: ${tab.id}`;
+  tabItem.querySelector('.tab_index').value = tab.index;
+  tabItem.querySelector('.tab_window_id').value = tab.windowId;
+  tabItem.querySelector('.tab_title').value = tab.title;
+  tabItem.querySelector('.tab_url').value = tab.url;
+  tabItem.querySelector('.tab_selected').checked = tab.selected;
+
+  tabItem.querySelector('.move_tab_button').addEventListener('click', () => {
+    moveTab(tab.id);
+  });
+  tabItem.querySelector('.refresh_tab_button').addEventListener('click', () => {
+    refreshTab(tab.id);
+  });
+  tabItem.querySelector('.update_tab_button').addEventListener('click', () => {
+    updateTab(tab.id);
+  });
+  tabItem.querySelector('.remove_tab_button').addEventListener('click', () => {
+    removeTab(tab.id);
+  });
+}
+
+function updateTabData(id) {
+  const retval = {
+    url: document.getElementById('url_' + id).value,
+    selected: document.getElementById('selected_' + id).value ? true : false
```

Comment:
I think this (and the same below) should use .checked since the value of a checkbox is always "on"?